### PR TITLE
Jenkins support: add dynamic plugins for RHDH

### PR DESCRIPTION
**Pull request by jkopriva**
image

### CODE REVIEW

**Comment by jkopriva** [comment]
Ci failed due to https://issues.redhat.com/browse/RHTAPINST-87

File: charts/rhtap-dh/templates/plugins.yaml [hunk 1]
```diff
@@ -33,6 +33,23 @@ data:
         package: ./dynamic-plugins/dist/backstage-plugin-techdocs
       - disabled: false
         package: ./dynamic-plugins/dist/backstage-plugin-kubernetes
+      - disabled: false
```
Comment (Roming22) [inline charts/rhtap-dh/templates/plugins.yaml:37]: The new plugins need to be gated based on the presence of the `rhtap-jenkins-integration`  secret. See how it's done for argocd.

File: charts/rhtap-dh/templates/app-config-content.yaml [hunk 2]
```diff
@@ -54,6 +54,12 @@ catalog:
       - Location
       - Template
       - API
+jenkins:
```
Comment (Roming22) [inline charts/rhtap-dh/templates/app-config-content.yaml:58]: The new plugins need to be gated based on the presence of the `rhtap-jenkins-integration`  secret. See how it's done for github.

File: charts/rhtap-dh/templates/extra-env.yaml [hunk 3]
```diff
@@ -51,3 +51,10 @@ data:
     QUAY__API_TOKEN: {{ $quaySecretData.token }}
 {{- end }}
     QUAY__URL: {{ $quaySecretData.url }}
+{{- $jenkinsSecretObj := (lookup "v1" "Secret" .Release.Namespace "rhtap-jenkins-integration") }}
+{{- $jenkinsSecretData := ($jenkinsSecretObj.data | default dict) }}
+{{- if $jenkinsSecretData.token }}
+    JENKINS__BASE__URL: {{ $jenkinsSecretData.baseUrl }}
```
Comment (Roming22) [inline charts/rhtap-dh/templates/extra-env.yaml]: JENKINS__BASE__URL -> JENKINS__BASEURL.

'_' is used to replace invalid characters.
'__' is used to separate nested fields.

**Review by Roming22 — APPROVED** [review]
lgtm

**Pull request merged**